### PR TITLE
Add basic array tests

### PR DESCRIPTION
**Pull request by mrocklin**
So, the in-memory one works great.  The out-of-memory one not so much.  I'm skipping for now because I suspect that this will kill things.

### CODE REVIEW

File: conftest.py [hunk 1]
```diff
@@ -15,3 +20,29 @@ def pytest_collection_modifyitems(config, items):
...
+@pytest.fixture
+def small_cluster(software):
+    with Cluster(
+        software=software,
+        account="dask-engineering",
+        n_workers=10,
+        worker_vm_types=["m5.large"],
+        scheduler_vm_types=["m5.large"],
+    ) as cluster:
+        yield cluster
```
Comment (gjoseph92) [inline conftest.py:42]: What shuts down the cluster when we're done?

File: conftest.py [hunk 2]
```diff
@@ -15,3 +20,29 @@ def pytest_collection_modifyitems(config, items):
...
+@pytest.fixture
+def software():
+    return os.environ.get(
+        "COILED_SOFTWARE_NAME",
+        f"dask-engineering/coiled_dist-py{sys.version_info[0]}{sys.version_info[1]}",
+    )
+
+
+@pytest.fixture
+def small_cluster(software):
```
Comment (gjoseph92) [inline conftest.py:34]: Do we want to allow multiple tests to use the same cluster? Should be be restarted each time? Recreated each time? Wondering about fixture scope.

File: conftest.py [hunk 3]
```diff
@@ -15,3 +20,29 @@ def pytest_collection_modifyitems(config, items):
...
+    with Cluster(
+        software=software,
+        account="dask-engineering",
+        n_workers=10,
+        worker_vm_types=["m5.large"],
+        scheduler_vm_types=["m5.large"],
```
Comment (gjoseph92) [inline conftest.py:40]:
```suggestion
        scheduler_vm_types=["m5.large"],
        wait_for_workers=True,
```
Debatable. Leaving it out increases chaos. Adding it increases determinism. Looks like the default is 0.3.

File: conftest.py (hunk 1 above)
Comment (mrocklin) [reply conftest.py:42]: The context manager hopefully.  Cluster.__exit__ should call Cluster.close

File: conftest.py (hunk 2 above)
Comment (mrocklin) [reply conftest.py:34]: Maybe.  This is what they're doing with the H2O benchmarks.  We'll want things clean, but maybe we can achieve that with restart, scale, and wait_for_workers.

The platform integration tests do a single cluster per test, and they just run all of the tests in parallel.  It's less efficient, but works ok.

I'm open here.  I think that we'll probably want to iterate.

File: conftest.py (hunk 3 above)
Comment (mrocklin) [reply conftest.py:40]: This repository is used both for stability and for performance testing.  My guess is that we're going to want to split this out pretty soon to different options, but for now I don't care that much.

File: conftest.py (hunk 2 above)
Comment (jrbourbeau) [reply conftest.py:34]: I'm suggesting we start with module-scoped cluster fixtures + function-scoped client fixtures with restarts over in https://github.com/coiled/coiled-runtime/pull/40. This is something that seems to work for our current needs and, as @mrocklin said, we can iterate as needed. 

+1 for parallelizing tests. Opening a separate issue for that.

Comment (mrocklin) [reply conftest.py:34]: In platform integration tests we asked pytest-xdist to parallelize between modules, but not within them.  That would work will with the module-scoped cluster fixtures.

**Comment by mrocklin** [comment]
@jrbourbeau ok to merge?

**Comment by ncclementi** [comment]
@mrocklin This looks good to me, I quickly check on the `test_rechunk_out_of_memory` I noticed that with the default instances and 10 workers and it is chugging. But I haven't tried with the `"m5.large"` I wonder if we want to keep skipping this or take a different approach. 

Screen Shot 2022-04-14 at 11 26 51 AM
Screen Shot 2022-04-14 at 11 27 46 AM

**Comment by mrocklin** [comment]
What sort of different approach would you propose?  Run it?  Change the test?

**Comment by ncclementi** [comment]
> What sort of different approach would you propose? Run it? Change the test?

1. If this is a test that is expected to run with the Cluster as is, but it doesn't we can skip it and open an issue. 
2. If the only reason why this doesn't run is that there is not enough memory then:
    a. change the test and make x smaller.
    b. use a different cluster configuration

cc: @jrbourbeau Do you have any comments? If not I'd be ok merging it as is and open a separate issue to sort out the reason why the test chugs.

**Comment by mrocklin** [comment]
> If the only reason why this doesn't run is that there is not enough memory then:
> a. change the test and make x smaller.
> b. use a different cluster configuration

Yeah, so my intention here is to run a computation that is uncomfortably large for Dask.  Dask should grow so that is can handle it gracefully.  Thus the skip + issue approach

**Review by jrbourbeau — APPROVED** [review]
Let's go with the skip + issue approach

**Pull request merged**